interview-questions.md

```diff
@@ -8,50 +8,50 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is the difference between a parameter and an argument?
 
-Your answer:
+Your answer: A parameter serves as a placeholder for values you're using to pass in an argument when you call that function. The parameters are named when you define a function.  
 
-Researched answer:
+Researched answer: Parameters are abstract variaible names that you are able to use when you define a function and serves as placeholders, also known as variable declaration in the function. However, arguments are the actual values or elements that you provide to the function that will be passed when you call on the function. The values that are passed will be stored and can be accesssed under the same parameters you assigned them to in the function.
 
 2. The JavaScript built in method .map() takes predefined parameters. What are they? Which are required and which are optional?
 
-Your answer:
+Your answer: The .map() method takes in 3 predfined parameters that are value, index, and array. Value is required because it contains the values of the elements of the array provided. Index and arr are optional.
 
-Researched answer:
+Researched answer: The .map() has value, index, and arr as the predefined parameters. Value is the only required predfined parameter because it contains the value of the array. Index and arr are optional. Index contains the index of the current element and arr contains the array where the map() function is applied. 
 
 3. What is the difference between map and filter?
 
-Your answer:
+Your answer: The .map() and .filter() method are quite similar but have some differences.The .map() method iterates each element in the array and returns the same length array once all the conditions are met. However, the .filter() method iterates over each element in the array but always want to return a new shorter array once the condtion is met.
 
-Researched answer:
+Researched answer: The .map() method creates a new array by calling a function on each values in the array. It will include the new desired data as a part of the new array. The .filter() method returns a new array of elements based on custom criteria that was set in the callback function. What they do have in common though is that they both return a new array.
 
 4. What is iteration?
 
-Your answer:
+Your answer: Iteration is a set of conditions that are being performed over and over again until the condtion is met. 
```

Comment:
Try expanding this answer by talking about the different types of iteration you have used as a developer.